app/scripts/controllers/transactions.js

```diff
@@ -221,9 +221,19 @@ angular.module('openhimConsoleApp')
         }
       });
     }, function(){ /* server error - could not connect to API to get channels */ });
-
-    // clients used for advanced filter dropdown
-    $scope.clients = Api.Clients.query();
+    
+    
+    var clientsIdNameMap = [];
+    
+    var clientsSuccess = function(clients) {
+      $scope.clients = clients;
+      // Create client map of id to name
+      $scope.clients.map(function(client) {
+        clientsIdNameMap[client._id] = client.name;
```

Comment:
Might be better to make this a scope variable, Then we would need the below function to be called for each client lookup. The the scope var can be used in the html instead of calling a function